httpbin/helpers.go

```diff
@@ -159,6 +159,19 @@ func parseBody(w http.ResponseWriter, r *http.Request, resp *bodyResponse) error
 		if err != nil && err != io.EOF {
 			return err
 		}
+	// in case of binary data, we encode it as base64 and return it as a data url
+	case strings.HasPrefix(ct, "image/png"):
+		fallthrough
+	case strings.HasPrefix(ct, "image/jpeg"):
+		fallthrough
+	case strings.HasPrefix(ct, "image/webp"):
+		fallthrough
+	case strings.HasPrefix(ct, "application/octet-stream"):
```

Comment:
I think this is a reasonable starting point, but I also got curious about httpbin's implementation.

It turns out [they do something fairly naive](https://github.com/postmanlabs/httpbin/blob/f8ec666b4d1b654e4ff6aedd356f510dcac09f83/httpbin/helpers.py#L85-L106):  If the incoming request body either a) cannot be decoded from utf8 or b) cannot be re-serialized to JSON, it is treated as binary data.

I wonder if we should do something similar here, instead of working with this fixed list of "probably binary" content-type headers?

Reply:
Maybe we should do both:  Keep this set of "probably binary" headers for which we _know_ we want return a base64-encoded `data:` url, but add a case to handle the situation where we've got unknown non-utf8 data in the incoming request body?

Reply:
I pushed a new commit that handles all non-text content this way.